include/dr/shp/algorithms/for_each.hpp

```diff
@@ -54,4 +54,18 @@ void for_each(Iter begin, Iter end, Fn &&fn) {
   for_each(dr::shp::par_unseq, begin, end, std::forward<Fn>(fn));
 }
 
+template <typename ExecutionPolicy, dr::distributed_iterator Iter,
+          std::integral I, typename Fn>
+void for_each_n(ExecutionPolicy &&policy, Iter begin, I n, Fn &&fn) {
+  auto end = begin;
+  rng::advance(end, n);
+  for_each(std::forward<ExecutionPolicy>(policy), begin, end,
+           std::forward<Fn>(fn));
+}
+
+template <dr::distributed_iterator R, std::integral I, typename Fn>
```

Comment:
convention in this file is to name distributed_iterator 'Iter' instead of `R` (used for ranges)